platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/collections/AbstractLongList.java

```diff
@@ -243,8 +243,35 @@ protected AbstractLongList(
      * @param fileChannel the file channel to read the list body from
      * @throws IOException if there was a problem reading the file
      */
-    protected abstract void readBodyFromFileChannelOnInit(String sourceFileName, FileChannel fileChannel)
-            throws IOException;
+    protected void readBodyFromFileChannelOnInit(String sourceFileName, FileChannel fileChannel) throws IOException {
+        if (minValidIndex.get() < 0) {
+            // Empty list, nothing to read
+            return;
+        }
+
+        final int totalNumberOfChunks = calculateNumberOfChunks(size.get());
+        final int firstChunkWithDataIndex = toIntExact(minValidIndex.get() / numLongsPerChunk);
+        final int minValidIndexInChunk = toIntExact(minValidIndex.get() % numLongsPerChunk);
+
+        for (int chunkIndex = firstChunkWithDataIndex; chunkIndex < totalNumberOfChunks; chunkIndex++) {
+            final int startIndex = (chunkIndex == firstChunkWithDataIndex) ? minValidIndexInChunk : 0;
+            C chunk = readChunkData(fileChannel, chunkIndex, startIndex);
```

Comment:
`startIndex` is great, what about the end index? First, the very last chunk may be shorter than `memoryChunkSize`. Second, there may be only one chunk in the list, in this case both start index and end index will be different from 0 and memoryChunkSize, respectively. Is there a test case for this?